tools/cli/src/mentat_cli/input.rs

```diff
@@ -80,7 +86,19 @@ impl InputReader {
 
         self.add_history(&line);
 
-        let cmd = try!(command(&self.buffer));
+        let cmd = match in_process_cmd {
+            Some(Command::Query(args)) => {
+                Command::Query(args + " " + &line)
+            },
+            Some(Command::Transact(args)) => {
+                Command::Transact(args + " " + &line)
+            },
+            _ => {
+                try!(command(&self.buffer))
```

Comment:
Prefer `...?` to `try!(...)`; it's the modern expression in Rust.  Are you really only doing this evaluation for the error result?  You should consume the result of any `Result`, using `let _ = ...` if needed; the compiler should be warning you that you're dropping a result on the floor here.